File: testcases/src/main/java/recursive/RecurciveRes.java
package recursive;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
@Path("recursive")
public class RecurciveRes {

    @GET
    public String getText() { return "foo"; }

    @Path("some_path")
    public RecurciveRes getSubResource() {
        return new RecurciveRes();
    }
}

File: testcases_kt/src/main/kotlin/recursive/RecurciveRes.kt
package recursive

import javax.ws.rs.GET
import javax.ws.rs.Path

@Path("recursive_kt")
class RecurciveRes {
    @get:GET
    val text: String
        get() = "foo"

    @get:Path("some_path")
    val subResource: RecurciveRes
        get() = RecurciveRes()
}
